Error compiling with the new version of Osquery 1.7.6

osquery versions `osqueryi version 1.7.6-3-g72df044`

@andrew-d  Can you please compile it with latest version of osquery? I am getting the following errors.

```
g++ -c -o osquery_profiles.o -g -I /home/chenone2316/GitRepos/osquery/include/ -DOSQUERY_BUILD_SDK_VERSION=1.7.6 -std=c++11 osquery_profiles.cpp
osquery_profiles.cpp: In function ‘osquery::Status parseProfile(const string&, const string&, Fn)’:
osquery_profiles.cpp:88:44: error: there are no arguments to ‘parsePlistContent’ that depend on a template parameter, so a declaration of ‘parsePlistContent’ must be available [-fpermissive]
   if (parsePlistContent(commandOutput, tree).ok()) {
                                            ^
osquery_profiles.cpp:88:44: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
osquery_profiles.cpp: In member function ‘virtual osquery::TableColumns ProfilesTablePlugin::columns() const’:
osquery_profiles.cpp:172:5: error: could not convert ‘{{"username", TEXT_TYPE}, {"type", TEXT_TYPE}, {"identifier", TEXT_TYPE}, {"display_name", TEXT_TYPE}, {"description", TEXT_TYPE}, {"organization", TEXT_TYPE}, {"verified", INTEGER_TYPE}, {"removal_allowed", INTEGER_TYPE}}’ from ‘<brace-enclosed initializer list>’ to ‘osquery::TableColumns {aka std::vector<std::tuple<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, osquery::ColumnType, osquery::ColumnOptions> >}’
     };
     ^
osquery_profiles.cpp: In member function ‘virtual osquery::TableColumns ProfileItemsTablePlugin::columns() const’:
osquery_profiles.cpp:229:5: error: could not convert ‘{{"username", TEXT_TYPE}, {"profile_identifier", TEXT_TYPE}, {"type", TEXT_TYPE}, {"identifier", TEXT_TYPE}, {"display_name", TEXT_TYPE}, {"description", TEXT_TYPE}, {"organization", TEXT_TYPE}, {"content", TEXT_TYPE}}’ from ‘<brace-enclosed initializer list>’ to ‘osquery::TableColumns {aka std::vector<std::tuple<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, osquery::ColumnType, osquery::ColumnOptions> >}’
     };
     ^
osquery_profiles.cpp: In instantiation of ‘osquery::Status parseProfile(const string&, const string&, Fn) [with Fn = ProfilesTablePlugin::generate(osquery::QueryContext&)::__lambda1; std::string = std::basic_string<char>]’:
osquery_profiles.cpp:123:61:   required from ‘osquery::Status iterateProfiles(osquery::QueryContext&, Fn) [with Fn = ProfilesTablePlugin::generate(osquery::QueryContext&)::__lambda1; osquery::QueryContext = osquery::QueryContext]’
osquery_profiles.cpp:203:6:   required from here
osquery_profiles.cpp:88:44: error: ‘parsePlistContent’ was not declared in this scope
   if (parsePlistContent(commandOutput, tree).ok()) {
                                            ^
osquery_profiles.cpp: In instantiation of ‘osquery::Status parseProfile(const string&, const string&, Fn) [with Fn = ProfileItemsTablePlugin::generate(osquery::QueryContext&)::__lambda2; std::string = std::basic_string<char>]’:
osquery_profiles.cpp:123:61:   required from ‘osquery::Status iterateProfiles(osquery::QueryContext&, Fn) [with Fn = ProfileItemsTablePlugin::generate(osquery::QueryContext&)::__lambda2; osquery::QueryContext = osquery::QueryContext]’
osquery_profiles.cpp:283:6:   required from here
osquery_profiles.cpp:88:44: error: ‘parsePlistContent’ was not declared in this scope

```

--- a/osquery_profiles.cpp
+++ b/osquery_profiles.cpp
@@ -1,4 +1,5 @@
 #include <osquery/sdk.h>
+#include <osquery/system.h>
 
 #include <iomanip>
 #include <sstream>

--- a/osquery_profiles.cpp
+++ b/osquery_profiles.cpp
@@ -158,17 +158,17 @@ class ProfilesTablePlugin : public TablePlugin {
  private:
   TableColumns columns() const {
     return {
-      {"username", TEXT_TYPE},
-      {"type", TEXT_TYPE},
-      {"identifier", TEXT_TYPE},
-      {"display_name", TEXT_TYPE},
-      {"description", TEXT_TYPE},
-      {"organization", TEXT_TYPE},
-      {"verified", INTEGER_TYPE},
-      {"removal_allowed", INTEGER_TYPE},
+      std::make_tuple("username", TEXT_TYPE, DEFAULT),
+      std::make_tuple("type", TEXT_TYPE, DEFAULT),
+      std::make_tuple("identifier", TEXT_TYPE, DEFAULT),
+      std::make_tuple("display_name", TEXT_TYPE, DEFAULT),
+      std::make_tuple("description", TEXT_TYPE, DEFAULT),
+      std::make_tuple("organization", TEXT_TYPE, DEFAULT),
+      std::make_tuple("verified", INTEGER_TYPE, DEFAULT),
+      std::make_tuple("removal_allowed", INTEGER_TYPE, DEFAULT),
 
       // TODO: add a 'version' column with the 'ProfileVersion' key?
-      //{"version", INTEGER_TYPE},
+      //std::make_tuple("version", INTEGER_TYPE, DEFAULT),
     };
   }
 
@@ -215,17 +215,17 @@ class ProfileItemsTablePlugin : public TablePlugin {
  private:
   TableColumns columns() const {
     return {
-      {"username", TEXT_TYPE},
-      {"profile_identifier", TEXT_TYPE},
-      {"type", TEXT_TYPE},
-      {"identifier", TEXT_TYPE},
-      {"display_name", TEXT_TYPE},
-      {"description", TEXT_TYPE},
-      {"organization", TEXT_TYPE},
-      {"content", TEXT_TYPE},
+      std::make_tuple("username", TEXT_TYPE, DEFAULT),
+      std::make_tuple("profile_identifier", TEXT_TYPE, DEFAULT),
+      std::make_tuple("type", TEXT_TYPE, DEFAULT),
+      std::make_tuple("identifier", TEXT_TYPE, DEFAULT),
+      std::make_tuple("display_name", TEXT_TYPE, DEFAULT),
+      std::make_tuple("description", TEXT_TYPE, DEFAULT),
+      std::make_tuple("organization", TEXT_TYPE, DEFAULT),
+      std::make_tuple("content", TEXT_TYPE, DEFAULT),
 
       // TODO: add a 'version' column with the 'PayloadVersion' key?
-      //{"version", INTEGER_TYPE},
+      //std::make_tuple("version", INTEGER_TYPE, DEFAULT),
     };
   }
 

